WCFDataService/Client/System/Data/Services/Client/Binding/BindingObserver.cs

```diff
@@ -399,6 +404,44 @@ internal void OnDataServiceCollectionChanged(object collection, NotifyCollection
             }
         }
 
+        internal void OnDataServiceCollectionBulkAdded(object collection, IEnumerable newItems)
+        {
+            Util.CheckArgumentNull(collection, "collection");
+            Util.CheckArgumentNull(newItems, "newItems");
+
+            Debug.Assert(BindingEntityInfo.IsDataServiceCollection(collection.GetType(), this.Context.Model), "We only register this event for DataServiceCollections.");
+#if DEBUG
```

Comment:
Correct. It is also there in the original OnDataServiceCollectionChanged, I'll remove the if DEBUG in both places.